src/components/Api.js

```diff
@@ -0,0 +1,131 @@
+export default class Api {
+  constructor({ baseURL, authToken }) {
+    this._baseURL = baseURL;
+    this._authToken = authToken;
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseURL}//users/me`, {
+      headers: {
+        authorization: this._authToken,
+      },
+    })
+      .then((res) => {
+        return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the **reference** for the method rather than the call. 

**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
Thank you for your comments and attention to detail! I created both functions and utilized the _checkResponse function but am not exactly sure how to use the _request function with all of the different functions in the API.
<img width="528" alt="Screen Shot 2024-08-10 at 1 22 36 AM" src="https://github.com/user-attachments/assets/be525887-af58-45b6-ab28-16d795874f09">

Reply:
> Thank you for your comments and attention to detail! I created both functions and utilized the _checkResponse function but am not exactly sure how to use the _request function with all of the different functions in the API.


This is the way you should use it


```js


  _request(url, options) {
    return fetch(url, options).then(this._checkResponse);
  }

  getUserInfo() {
    return this._request(`${this._baseURL}/users/me`, {
      headers: this._headers,
    });
  }

  getInitialCards() {
    return this._request(`${this._baseURL}/cards`, {
      headers: this._headers,
    });
  }

  getAppInfo() {
    return Promise.all([this.getUserInfo(), this.getInitialCards()]);
  }

  editProfile({ name, about }) {
    return this._request(`${this._baseURL}/users/me`, {
      method: "PATCH",
      headers: this._headers,
      body: JSON.stringify({ name, about }),
    });
  }

```